shared/Dictionary.lua

```diff
@@ -0,0 +1,51 @@
+Dictionary = class(ValueStorage)
```

Comment:
We should refactor this code like this:
1. LanguageEnum class to define languages
2. PhraseEnum class to define phrases
2. Phrases class as the interface for any gamemode logic

I see it working like this:
```
LanguageEnum = immediate_class(Enum)

function LanguageEnum:__init()
  self:EnumInit()
  
  self.English = 1
  self.Spanish = 2

  self:SetDescription(self.English, "English")
  self:SetDescription(self.Spanish, "Spanish")
end

LanguageEnum = LanguageEnum()
```

```
PhraseEnum = immediate_class(Enum)

function PhraseEnum:__init()
  self:EnumInit()

  self.WelcomeText = 1
end

PhraseEnum = PhraseEnum()
```

```
Phrases = class()

function Phrases:__init()
  self.translations = {
    [PhraseEnum.WelcomeText] = {
        [LanguageEnum.English] = "Welcome to the server!",
        [LanguageEnum.Spanish] = "Bienvenidos!"
    },
    -- etc
  }
end

function Phrases:GetPhrase(phrase_enum)
  local language = NapiConfig.Language
  local translation = self.translations[phrase_enum]
  assert(phrase_enum ~= nil, "Phrase does not have any translation")
  assert(translation[language] ~= nil, "Phrase does not have a translation in " .. LanguageEnum:GetDescription(language))
  return translation[language]
end

Phrases = Phrases()
```

Now we can do `local text = Phrases:GetPhrase(PhraseEnum.WelcomeText)`

Reply:
Didn't better will be to create language enums and phrases in different files (not in class)?(one for each language in special directory)

Reply:
This will at least make it easier to add phrases in different languages to the repository.

Reply:
Yes - it would be good to put all the phrases for each language in its own file. You could create a directory in napi/shared and come up with a way to do this.
Also, we are not creating the enums in the `Phrases` class. We are just creating a table so we can look up a translation for a Phrase in a given Language. The enums are good practice and should be kept for readability and for the sake of providing a unique id to each Phrase and Language.

Reply:
Also such system where all is Enum it's a wildness for me if we want to make it simple to use by develoler in his own gamemode. If languages are initialized in only one place, it creates certain limitations.
Not better to name it to work so?:
-- russian.lua
Russian = Language("Russian")

Russian.Phrase("greetings", "Добро пожаловать!")
Russian.Phrase("kicked", "Вы были кикнуты с сервера")

-- config.lua
NapiConfig:SetLanguage("Russian")
-- script.lua
Language:GetPhrase("kicked")

it seems to be the most simplified and dev-friendly variant. If SetLanguage not exists than we take English Language, if phrase not exists then we took it in English.